packages/playwright-core/src/server/registry/browserFetcher.ts

```diff
@@ -37,7 +37,8 @@ export async function downloadBrowserWithProgressBar(title: string, browserDirec
     return false;
   }
 
-  const zipPath = path.join(os.tmpdir(), downloadFileName);
+  const downloadFolder = fs.mkdtempSync(path.join(os.tmpdir(), 'playwright-download-'));
```

Comment:
- You are in async context, use async fs functions
- Where is this folder cleaned up?

Reply:
Should be resolved now, folder is being cleaned up near where we unlink the zipPath